### ISSUE TITLE
onnx runtime error 6 when using ORT 1.16.1 with TRT and CUDA EP

### ISSUE THREAD

**Issue by Tabrizian** [microsoft/onnxruntime#18065]
### Describe the issue

When running inferences with Triton's ORT backend with ORT 1.16.1 with CUDA EP and TRT EP, we have ran into the issue below:

```
inference failed: [StatusCode.INTERNAL] onnx runtime error 6: /workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_call.cc:121 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_call.cc:114 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] CUBLAS failure 1: invalid argument ; GPU=0 ; hostname=fb56312c2595 ; file=/workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_stream_handle.cc ; line=57 ; expr=cublasSetStream(cublas_handle_, stream);
```

Note that the ORT backend works fine when using the ORT 1.16.0.

### To reproduce

1. Compile ORT backend with ORT 1.16.1
2. Download a ResNet model and enable TRT execution provider by adding the line below to the model configuration:

```
optimization { execution_accelerators { gpu_execution_accelerator : [ { name : "tensorrt"} ] } }
```

3. Run an inference using the [image_client](https://github.com/triton-inference-server/client/blob/23fa9b11321c379427920d8deca7c2535ed05219/src/python/examples/image_client.py) and the [vulture.jpg](https://github.com/triton-inference-server/server/blob/main/qa/images/vulture.jpeg) image.

4. Observe the error.

### Platform

Linux

### OS Version

Ubuntu 22.04

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.16.1

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

CUDA, TensorRT

### Execution Provider Library Version

CUDA 12.2

**Comment by jywu-msft** [comment]
this is a known issue and will be fixed in 1.16.2

**Comment by JulienTheron** [comment]
I just went to create a new issue about this.
When using ORT directly, the problem occurs when using a user-created compute stream with TensorRT. Does not happen with 1.16.0.

**Comment by chilo-ms** [comment]
Yes, that's right, it's a regression for 1.16.1.
Here is the fix https://github.com/microsoft/onnxruntime/commit/28c1944561cbd21256368943e452ac304fb8f582 and will be included in 1.16.2

**Comment by Tabrizian** [comment]
Thanks for the quick response. I'll be closing this issue.
